File: NUMBERS/9. Palindrome Number.cpp
#include <iostream>
using namespace std;

class Solution {
public:
    bool isPalindrome(int x) {

        // Negative numbers are not palindrome
        if (x < 0) {
            return false;
        }

        int original = x;
        int rev = 0;

        // Reverse the number
        while (x > 0) {

            int digit = x % 10;

            rev = rev * 10 + digit;

            x = x / 10;
        }

        // Check original and reversed number
        return original == rev;
    }
};

int main() {

    Solution s;

    int x = 121;

    if (s.isPalindrome(x)) {
        cout << "Palindrome";
    }
    else {
        cout << "Not Palindrome";
    }

    return 0;
}

File: NUMBERS/9. Palindrome Number.js
function isPalindrome(x) {
    if (x < 0) {
        return false;
    }
    // Convert number to string
    const str = x.toString();
    const n = str.length;

    // Check characters from both ends
    for (let i = 0; i < Math.floor(n / 2); i++) {
        if (str[i] !== str[n - 1 - i]) {
            return false;
        }
    }

    return true;
}

console.log(isPalindrome(121));  
console.log(isPalindrome(-121)); 
console.log(isPalindrome(10));   